### DISCUSSION TITLE
How to reduce real-time audio and video stream forwarding delay？

### DISCUSSION THREAD

**Discussion by youfeed** [uNetworking/uWebSockets.js#1030] [Q&A]
My business uses pub/sub format to forward calls between both parties.

How to get only the latest push on the sub side and allow discarding part of the old stream to ensure the lowest delay
```
  app.ws('/pull/:uuid',{
    compression: uWS.SHARED_COMPRESSOR,
    idleTimeout: 60,
    maxBackpressure: 64 * 1024 * 1024,
    maxPayloadLength: 64 * 1024 * 1024,
    sendPingsAutomatically:true
   .........
})
```

**Comment by uNetworkingAB** [comment]
Websockets are TCP so already you have kind of the wrong protocol for video calls. But you can lower max backpressure and TCP socket send buffer sizes to allow the lib to skip old data

**Comment by youfeed** [comment]
I'll try
```
maxBackpressure: 1024,
```
wait me back

**Comment by uNetworkingAB** [comment]
You also need to set the kernel TCP send buffer size small

**Comment by youfeed** [comment]
![image](https://github.com/uNetworking/uWebSockets.js/assets/22223510/3d82a60f-8b95-4c4e-9b80-b767ff0b50f3)
ws:/push/:uuid Used to push binary messages to the pull channel
ws:/pull/:uuid Specifically used to subscribe to channels

He is slower now
